### PR TITLE
Image usage conversions

### PR DESCRIPTION
**Pull request by kvark**

### CODE REVIEW

File: libportability-gfx/src/conv.rs [hunk 1]
```diff
@@ -16,6 +16,7 @@ pub fn limits_from_hal(limits: Limits) -> VkPhysicalDeviceLimits {
         maxFramebufferHeight: limits.max_texture_size as _, //TODO
         maxTexelBufferElements: limits.max_texture_size as _, //TODO
         maxTessellationPatchSize: limits.max_patch_size as _,
+        maxPushConstantsSize: 0x100, //TODO
```
Comment (msiglreith) [inline libportability-gfx/src/conv.rs]: That's quite alot. The required minimum is 128 (0x80) which we should also expose here. Even some Vulkan drivers don't support more.

File: libportability-gfx/src/conv.rs [hunk 2]
```diff
@@ -268,6 +269,31 @@ pub fn map_image_usage(usage: VkImageUsageFlags) -> image::Usage {
     flags
 }
 
+pub fn map_image_usage_from_hal(usage: image::Usage) -> VkImageUsageFlags {
```
Comment (msiglreith) [inline libportability-gfx/src/conv.rs]: `mem::transmute` not possible here?

File: libportability-gfx/src/conv.rs [hunk 3]
```diff
@@ -450,12 +476,6 @@ pub fn map_stage_flags(stages: VkShaderStageFlags) -> pso::ShaderStageFlags {
     if stages & VkShaderStageFlagBits::VK_SHADER_STAGE_COMPUTE_BIT as u32 != 0 {
         flags |= pso::ShaderStageFlags::COMPUTE;
     }
-    if stages & VkShaderStageFlagBits::VK_SHADER_STAGE_ALL_GRAPHICS as u32 != 0 {
```
Comment (grovesNL) [inline libportability-gfx/src/conv.rs]: Why don't we need these anymore?

File: libportability-gfx/src/conv.rs [hunk 4]
```diff
@@ -268,6 +269,31 @@ pub fn map_image_usage(usage: VkImageUsageFlags) -> image::Usage {
     flags
 }
 
+pub fn map_image_usage_from_hal(usage: image::Usage) -> VkImageUsageFlags {
+    let mut flags = 0u32;
```
Comment (grovesNL) [inline libportability-gfx/src/conv.rs]: Why can't we simply transmute these like we do in the Vulkan backend?

File: libportability-gfx/src/conv.rs (hunk 3 above)
Comment (msiglreith) [reply libportability-gfx/src/conv.rs]: These are just combinations of the flags above.

Comment (grovesNL) [reply libportability-gfx/src/conv.rs]: Ah ok

**Comment by kvark** [comment]
Great feedback, thanks guys!
Addressed now.
bors r=msiglreith

**Pull request merged**